src/main/kotlin/no/nav/familie/ef/sak/service/BrevService.kt

```diff
@@ -1,33 +1,43 @@
 package no.nav.familie.ef.sak.service
 
+import no.nav.familie.ef.sak.api.dto.BrevRequest
 import no.nav.familie.ef.sak.brev.BrevClient
+import no.nav.familie.ef.sak.repository.VedtaksbrevRepository
+import no.nav.familie.ef.sak.repository.domain.Vedtaksbrev
+import no.nav.familie.ef.sak.service.steg.StegType
 import org.springframework.stereotype.Service
+import java.util.*
 
 @Service
-class BrevService(private val brevClient: BrevClient) {
-
-    fun lagBrev(): ByteArray {
-        val body = """
-            {
-                "flettefelter": {
-                    "navn": [
-                        "Navn Navnesen"
-                    ],
-                    "fodselsnummer": [
-                        "1123456789"
-                    ],
-                    "dato": [
-                        "01.01.1986"
-                    ]
-                }
-            }
-        """.trimIndent()
-
-
-        /*
-        * Logikk for brevgenering her
-        */
-
-        return brevClient.genererBrev("brukesIkke", "brukesIkke", body)
+class BrevService(private val brevClient: BrevClient,
+                  private val brevRepository: VedtaksbrevRepository,
+                  private val behandlingService: BehandlingService,
+                  private val fagsakService: FagsakService,
+                  private val personService: PersonService) {
+
+    fun lagBrev(behandlingId: UUID): ByteArray {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        val fagsak = fagsakService.hentFagsak(behandling.fagsakId)
+        val person = personService.hentSøker(fagsak.hentAktivIdent())
+        val navn = person.navn.get(0)
+
+        val request = BrevRequest(navn = "${navn.fornavn} ${navn.etternavn}", ident = fagsak.hentAktivIdent())
+
+        val brev = brevRepository.insert(Vedtaksbrev(behandlingId = behandlingId,
+                                                     brevRequest = request,
+                                                     steg = behandling.steg,
+                                                     pdf = brevClient.genererBrev("bokmaal",
+                                                                                  "testDokument",
+                                                                                  request)))
+        return brev.pdf
+    }
+
+    fun hentBrev(behandlingId: UUID): Vedtaksbrev? {
+        return brevRepository.findByBehandlingId(behandlingId).first()
+    }
+
+    fun hentBrevForTotrinnskontroll(behandlingId: UUID): Vedtaksbrev? {
+        return brevRepository.findByBehandlingId(behandlingId).firstOrNull { it.steg === StegType.BESLUTTE_VEDTAK }
```

Comment:
Litt usikker på denne, den er ikke brukt.

Hva skjer hvis man går 2 runder i totrinnskontroll, hvordna håndteres oppdatering av ett brev?

Reply:
Vi har foreløpig ikke tatt høyde for at totrinnskontroll kan bli underkjent, så det er noe som må fikses. En mulig MVP-løsning kan være å slette begge brev ved underkjent totrinnskontroll... men skal tenke litt mer på den.

Reply:
Høres fint ut det :)